funsor/numpy.py

```diff
@@ -7,6 +7,45 @@
 from funsor.util import quote
 
 
+def einsum(equation, *operands):
```

Comment:
adapted from Pyro [torch_log](https://github.com/pyro-ppl/pyro/blob/dev/pyro/ops/einsum/torch_log.py)

Reply:
As I mentioned above, this should go in a separate file like funsor/einsum/numpy_log.py, in particular because we might want to add a linear (sum-product) or a tropical (max,sum) version in the future, and we'll want to set e.g. `backend=funsor.einsum.numpy_tropical` for those cases. Note these are not user-facing; they are only consumed by `opt_einsum`.